OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyUtils.cs

```diff
@@ -230,9 +230,15 @@ public SlotDropDownOption(string title, string order, Func<bool> selected)
 		public static Dictionary<CPos, SpawnOccupant> GetSpawnOccupants(Session lobbyInfo, MapPreview preview)
 		{
 			var spawns = preview.SpawnPoints;
-			return lobbyInfo.Clients
+			var occupants = lobbyInfo.Clients
```

Comment:
Mixing LINQ and manual enumeration isn't great.

It would be better if we either stick completely with LINQ, something like:
```csharp
return lobbyInfo.Clients
	.Where(c => (c.SpawnPoint - 1 >= 0) && (c.SpawnPoint - 1 < spawns.Length))
	.Select(c => new SpawnOccupant(c))
	.Concat(lobbyInfo.DisabledSpawns
		.Where(d => d - 1 < spawns.Length)
		.Select(d => new SpawnOccupant(d)))
	.ToDictionary(c => spawns[c.SpawnPoint - 1], c => c);
```

or define the empty dictionary and then populate it using two `foreach` loops.